### ISSUE TITLE
Improve the confusing error [3000] message

### ISSUE THREAD

**Issue by murrayc3** [xrootd/xrootd#2157]
Under certain error conditions XRootD generates the following `[3000]` error message: 
```
INFO    Tue, 19 Dec 2023 10:03:44 +0100; [1702976624039] BOTH xroot    TRANSFER:TYPE    3rd pull
...
INFO    Tue, 19 Dec 2023 10:03:47 +0100; Gfal2: Event triggered: BOTH xroot TRANSFER:EXIT Job finished, [ERROR] Server responded with an error: [3000]  [[x]root[s]://<host>[:<port>]/]<path> | -
...
ERR     Tue, 19 Dec 2023 10:03:47 +0100; Recoverable error: [22] Error on XrdCl::CopyProcess::Run(): [ERROR] Server responded with an error: [3000]  [[x]root[s]://<host>[:<port>]/]<path> | -\x00
```
The error message is incomprehensible.

The above logs were generated by an FTS server trying to copy a file using the XRoot protocll and an EOS storage endpoint.  Please could the error message give more XRootD context and not display the generic XRoot URL followed a strange escaped hexadecimal.  Could the message state what XRootD what trying to do at the time?

The problem with the above error message is that end users do not understand it.  They then contact the FTS team who also do not understand it.  The storage endpoint is finally contacted and hopefully, fingers crossed they understand what is wrong in a timely fashion.  We need to reduce the amount of work created by incomprehensible error messages.

**Comment by amadio** [comment]
I agree @murrayc3, there's definitely room for improvement in the errors that XRootD emits. We will have a campaign to improve not only this one, but several other similarly incomprehensible messages.
